redis/redis.go

```diff
@@ -28,19 +29,19 @@ import (
 var ErrNotConfigured = errors.New("Redis is not configured")
 
 type RedisConfig struct {
-	Address           string   `yaml:"address,omitempty"`
-	Username          string   `yaml:"username,omitempty"`
-	Password          string   `yaml:"password,omitempty"`
-	DB                int      `yaml:"db,omitempty"`
-	UseTLS            bool     `yaml:"use_tls,omitempty"`
-	MasterName        string   `yaml:"sentinel_master_name,omitempty"`
-	SentinelUsername  string   `yaml:"sentinel_username,omitempty"`
-	SentinelPassword  string   `yaml:"sentinel_password,omitempty"`
-	SentinelAddresses []string `yaml:"sentinel_addresses,omitempty"`
-	ClusterAddresses  []string `yaml:"cluster_addresses,omitempty"`
-	DialTimeout       int      `yaml:"dial_timeout,omitempty"`
-	ReadTimeout       int      `yaml:"read_timeout,omitempty"`
-	WriteTimeout      int      `yaml:"write_timeout,omitempty"`
+	Address           string      `yaml:"address,omitempty"`
+	Username          string      `yaml:"username,omitempty"`
+	Password          string      `yaml:"password,omitempty"`
+	DB                int         `yaml:"db,omitempty"`
+	TLS               xtls.Config `yaml:"tls,omitempty"`
```

Comment:
this will break existing users that do use TLS. I would recommend the following:

- keep `use_tls`, but mark it as deprecated
- use a pointer type `*xtls.Config` for `tls`
- when reading config, if `tls` is not set, then defer to `use_tls` value

then in the next major version, we could then remove it.

Reply:
done